### PR TITLE
Makes warehouses have lights turned off from roundstart. Also adds mapping helper for turning off lightswitch.

### PR DESCRIPTION
**Pull request by Helg2**
## About The Pull Request
Warehouse is roundstart closed with shutters and doesn't meant to spawn in, as far as i know. Logic is simple: why do you have turned on lights in a room that is barely used.

Basically every warehouse with a lightswitch now has lights turned off. Northstar has all the lights broken in it, so i don't know if i should add one to it.

I added a lightswitch on Birdshot so it looks like this:
image
And a lightswitch to Tram's warehouse:
image
Also added some lights on Metastation so it looks like this:
image
image

I made lights to turn off with mapping helper. Suggestions accepted.

I also forgot to make a fridge free accessed on the Beach, so i'm changing it with this pr.
## Why It's Good For The Game
Less electricity spending for no4raisin.
## Changelog
:cl:
add: [Birdshot], [Tram] Added a lightswitch to warehouse.
add: [Birdshot], [Deltastation], [Metastation], [Tram] and [Icebox] warehouses now have lights off roundstart.
fix: [Metastation] Warehouse now has lights. Finally.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Are there any cases where the turned_off type isn't used?
Would it be more useful to make a mapping helper that on initialize finds the area of its turf and toggles the lightswitch var to FALSE?

**Comment by Helg2** [comment]
> Are there any cases where the turned_off type isn't used?

Northstar and downstreams maybe?
> Would it be more useful to make a mapping helper that on initialize finds the area of its turf and toggles the lightswitch var to FALSE?

I'll try to do it.

**Comment by Jolly-66** [comment]
Mapping helper would be very nice to have ngl

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -546,6 +546,27 @@
...
+		return INITIALIZE_HINT_QDEL
+	payload()
+	qdel(src)
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/mapping_helpers/turn_off_lights/proc/payload()
+	var/area/needed_area = get_area(src)
+	if(!needed_area.lightswitch)
+		log_mapping("[src] at [AREACOORD(src)] [(needed_area.type)] tried to turn lights off but they are already off!")
+	needed_area.lightswitch = FALSE
```
Comment (LemonInTheDark) [inline code/modules/mapping/mapping_helpers.dm:568]: Should this be a stacktrace?

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -546,6 +546,27 @@
...
+/obj/effect/mapping_helpers/turn_off_lights/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		log_mapping("[src] spawned outside of mapload!")
+		return INITIALIZE_HINT_QDEL
+	payload()
+	qdel(src)
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/mapping_helpers/turn_off_lights/proc/payload()
```
Comment (LemonInTheDark) [inline code/modules/mapping/mapping_helpers.dm:564]: you should name this something that better describes it

File: code/modules/mapping/mapping_helpers.dm [hunk 3]
```diff
@@ -546,6 +546,27 @@
...
+	icon_state = "lights_off"
+	late = TRUE
+
+/obj/effect/mapping_helpers/turn_off_lights/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		log_mapping("[src] spawned outside of mapload!")
+		return INITIALIZE_HINT_QDEL
+	payload()
+	qdel(src)
```
Comment (LemonInTheDark) [inline code/modules/mapping/mapping_helpers.dm:561]: 🗞️ do not delete atoms inside Initialize, just return `INITIALIZE_HINT_QDEL`

File: code/modules/mapping/mapping_helpers.dm [hunk 4]
```diff
@@ -546,6 +546,27 @@
...
+	return INITIALIZE_HINT_LATELOAD
```
Comment (LemonInTheDark) [inline code/modules/mapping/mapping_helpers.dm:562]: Is this needed? you don't do anything in LateInitialize here

File: code/modules/mapping/mapping_helpers.dm [hunk 5]
```diff
@@ -546,6 +546,27 @@
 		log_mapping("[src] at [AREACOORD(src)] [(apc_area.type)] tried to set [target]'s charge to 100 but it's already at 100!")
 	target.full_charge = TRUE
 
+//Used to turn off lights in areas.
+/obj/effect/mapping_helpers/turn_off_lights
+	name = "area lights off helper"
+	icon_state = "lights_off"
+	late = TRUE
```
Comment (LemonInTheDark) [inline code/modules/mapping/mapping_helpers.dm:553]: ditto

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/mapping/mapping_helpers.dm (hunk 4 above)
Comment (Helg2) [reply code/modules/mapping/mapping_helpers.dm:562]: copypaste issues xd

File: code/modules/mapping/mapping_helpers.dm (hunk 2 above)
Comment (Helg2) [reply code/modules/mapping/mapping_helpers.dm:564]: `turn_off_lights_in_area` or something? no idea tbh

File: code/modules/mapping/mapping_helpers.dm (hunk 1 above)
Comment (Helg2) [reply code/modules/mapping/mapping_helpers.dm]: What's stacktrace?

File: code/modules/mapping/mapping_helpers.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/mapping/mapping_helpers.dm]: The proc I mean, sorry shoulda been more clear the type was named fine

File: code/modules/mapping/mapping_helpers.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/mapping/mapping_helpers.dm]: it logs a bit of text as an error like CRASH, but it allows the proc to continue execution.
It's good for doing something suspect under protest. You can raise a warning and then do the thing anyway, on the off chance it's the desired outcome

File: code/modules/mapping/mapping_helpers.dm (hunk 2 above)
Comment (Helg2) [reply code/modules/mapping/mapping_helpers.dm]: This is fine.

File: code/modules/mapping/mapping_helpers.dm (hunk 1 above)
Comment (Helg2) [reply code/modules/mapping/mapping_helpers.dm]: I'm not sure if this is a good idea tho

**Pull request merged**